x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -610,9 +600,12 @@ private AggGroupingFailure(List<String> expectedGrouping) {
         }
 
         @Override
-        protected LogicalPlan rule(LogicalPlan plan) {
+        protected LogicalPlan rulePlan(LogicalPlan plan) {
+            if (plan.resolved()) {
```

Comment:
I don't think this is needed anymore - see `AnalyzeRule` which takes care of skipping if the node is resolved and `skipResolved` is true (default).